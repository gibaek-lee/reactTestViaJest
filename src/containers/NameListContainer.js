import NameList from '../components/NameList';
import { connect } from 'react-redux';

const mapStateToProps = (state) => ({
  nameList: state.nameData.nameList
});

const NameListContainer = connect(
  mapStateToProps, null
)(NameList);

export default NameListContainer;
