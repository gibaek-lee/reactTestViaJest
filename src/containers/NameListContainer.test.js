import React from 'react';
import { mount } from 'enzyme';//enzyme full rendering
import NameListContainer from './NameListContainer';
import configureMockStore from 'redux-mock-store';

describe('NameListContainer', () => {
  let component = null;
  const mockStore = configureMockStore();

  let store = mockStore({
    nameData: {
      name: '',
      nameList: []
    }
  });

  it('renders properly', () => {
    component = mount(<NameListContainer store={store}/>)
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
