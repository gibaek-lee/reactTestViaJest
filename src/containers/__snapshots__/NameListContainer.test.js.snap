// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameListContainer matches snapshot 1`] = `
<Connect(NameList)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <NameList
    dispatch={[Function]}
    nameList={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div />
  </NameList>
</Connect(NameList)>
`;
